### PR TITLE
Feature/error handling message

### PR DESCRIPTION
**Pull request by aliroemhildt**
## Checklist
- [ ] Tests written where required
- [ ] All tests passing
- [ ] Unused code/prys/etc deleted
Testing Coverage at <percentage>%

## Changes Made
  
What (if any) features are you implementing?
- added error handling for incorrect urls and all graphQL requests (let me know if I missed anything, but I believe I covered all of them)
- created Error component

What (if anything) did you refactor?
- moved some queries from the component file to the queries file in order to clean up the component code
- adjusted layout of winner on landing page
- added loading spinner to landing page 

Were there any issues that arose?
- It's hard to test for 400 and 500 errors, but I believe the handling in the error component should cover that 

Is there anything that you need from your teammate?
- Need to make sure we are covering error handling on form submission

Any other comments, questions, or concerns?

### CODE REVIEW

File: src/components/Error/Error.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  return (
+    <div className="error-container">
+      <p className="error-message">{ message }</p>
+      { homeButton }
+    </div>
+  )
+}
+
+export default Error;
```
Comment (mr-ryan12) [inline src/components/Error/Error.js:38]: Really great way to conditionally render the button element! Super clean and easy to read!

File: src/components/LandingPage/LandingPage.js [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
+import Error from '../Error/Error';
+import "./LandingPage.scss"
 
 const LandingPage = () => {
   const [winner, setWinner] = useState(null);
 
   useEffect(() => {
     refetch()
-      .then(response => {
-        console.log(response.data.getWinner)
```
Comment (mr-ryan12) [inline src/components/LandingPage/LandingPage.js:13]: Nice catch!

File: src/components/LandingPage/LandingPage.js [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
+    </div>
+  
+  if (loading) {
+    return (
+      <div className="loading-container">
+        <UploadingSpinner />
+      </div>
+    );
+  }
+  if (error) return <Error error={error}/>
```
Comment (mr-ryan12) [inline src/components/LandingPage/LandingPage.js:32]: This looks great!

File: src/components/SubmissionDetailsPage/SubmissionDetailsPage.js [hunk 4]
```diff
@@ -65,7 +28,7 @@ const SubmissionDetailsPage = ({ user }) => {
     <Modal name={data.getSubmission.name} setShowModal={setShowModal} id={id}/>
   
   if (loading) return <p>loading...</p>
```
Comment (mr-ryan12) [inline src/components/SubmissionDetailsPage/SubmissionDetailsPage.js:30]: Did we want to use the `<UploadingSpinner/>` here as well?

**Review by mr-ryan12 — APPROVED** [review]
Really well done Ali! Thank you for taking initiative to get this done! Very much appreciated. I noticed there is an additional test failing in the `admin_spec.js` file, so I can look into that since I'm refactoring that file at the moment.

File: src/components/LandingPage/LandingPage.js [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
   const { loading, error, data, refetch } = useQuery(GET_WINNER)
   
   const displayWinner = winner &&
-    <p className="winner">{data.getWinner.name} is the winner!!!!!!!</p>
-
-  return(
+    <div className="winner-container">
+      <p className="winner-title">Our Troubadour 2022 winner is:</p>
+      <p className="winner-name">{data.getWinner.name}</p>
+    </div>
```
Comment (mr-ryan12) [inline src/components/LandingPage/LandingPage.js:23]: Ah, okay this makes sense why there is an additional test failing in `admin_spec.js`. I'll be sure to update the test to target the new `className=winner-container`

File: src/App.js [hunk 6]
```diff
@@ -19,9 +20,8 @@ function App() {
           <Route exact path='/' element={<LandingPage />} />
           <Route exact path='/submissions' element={<AdminPage user={user}/>} />
           <Route exact path='/form' element={<SubmissionForm />} />
-          <Route path='/submissions/:id' element={<SubmissionDetailsPage user={user}/>}
-          />
-          {/* <Route path='/*' element={<Error />} /> */}
+          <Route path='/submissions/:id' element={<SubmissionDetailsPage user={user}/>} />
+          <Route path='/*' element={<Error />} />
```
Comment (oacortez) [inline src/App.js:24]: Nice use of error handling here for path routing!

File: src/components/Error/Error.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { Link } from 'react-router-dom';
```
Comment (oacortez) [inline src/components/Error/Error.js:1]: Thank you for making this error component to handle errors

File: src/components/SubmissionDetailsPage/SubmissionDetailsPage.js [hunk 8]
```diff
@@ -45,15 +17,6 @@ const SubmissionDetailsPage = ({ user }) => {
     variables: { id: id }
   });
 
-  // const [ selectWinner, { loadingM, errorM, dataM }] = useMutation(UPDATE_WINNER, {
```
Comment (oacortez) [inline src/components/SubmissionDetailsPage/SubmissionDetailsPage.js:48]: Thank you for removing unused code for better readability!

File: src/components/SubmissionDetailsPage/SubmissionDetailsPage.js (hunk 4 above)
Comment (oacortez) [inline src/components/SubmissionDetailsPage/SubmissionDetailsPage.js:30]: I agree with Ryan maybe we can use our built loading component there!

**Review by oacortez — APPROVED** [review]
Thank you for making this branch Ali! I see a new component and logic to handle errors for our users. Everything looks great to me. I notice a failed ci test. Hopefully that is an easy fix.

**Review by Echory — APPROVED** [review]
Thank you for doing this, very clean!

File: src/components/SubmissionDetailsPage/SubmissionDetailsPage.js (hunk 4 above)
Comment (aliroemhildt) [reply src/components/SubmissionDetailsPage/SubmissionDetailsPage.js:30]: Good catch, I'd say yes! I think we should use it anywhere we have a loading gql request

File: src/components/LandingPage/LandingPage.js (hunk 5 above)
Comment (aliroemhildt) [reply src/components/LandingPage/LandingPage.js:23]: thanks for making that fix!

**Pull request merged**